saleor/payment/migrations/0041_add_calculation_transaction_events.py

```diff
@@ -1,37 +1,157 @@
+from decimal import Decimal
+
 from django.db import migrations
-from django.apps import apps as registry
-from django.db.models.signals import post_migrate
-
-from .tasks.saleor3_12 import (
-    create_event_for_authorized,
-    create_event_for_canceled,
-    create_event_for_charged,
-    create_event_for_refunded,
-    create_event_for_authorized_task,
-    create_event_for_canceled_task,
-    create_event_for_charged_task,
-    create_event_for_refunded_task,
+from django.db.models import (
+    Case,
+    Exists,
+    F,
+    Func,
+    OuterRef,
+    Subquery,
+    Q,
+    When,
 )
-from ... import __version__
 
 
-def create_transaction_events(apps, schema_editor):
-    def on_migrations_complete(sender=None, **kwargs):
-        create_event_for_authorized_task.delay()
-        create_event_for_canceled_task.delay()
-        create_event_for_charged_task.delay()
-        create_event_for_refunded_task.delay()
+def create_transaction_events(transaction_event, transactions_qs, type):
```

Comment:
Is it safe to not use batches here?

Reply:
Yes, as transactions are in preview feature.